connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -207,7 +208,7 @@ public ConnectorStateInfo connectorStatus(String connName) {
 
         Collections.sort(taskStates);
 
-        return new ConnectorStateInfo(connName, connectorState, taskStates);
+        return new ConnectorStateInfo(connName, connectorState, taskStates, connectorTypeForClass(connName));
```

Comment:
this is still trying to do the incorrect lookup -- passing in the connector name, i.e. "my-sink", but the method it is being passed to seems to expect the class, i.e. "FooSinkConnector". That makes requests to the status endpoint still fail:

```
$ curl http://localhost:8083/connectors/local-file-sink/status | jq
{
  "error_code": 500,
  "message": "Failed to find any class that implements Connector and which name matches local-file-sink, available connectors are: PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSinkConnector, name='org.apache.kafka.connect.file.FileStreamSinkConnector', version='1.0.0-SNAPSHOT', encodedVersion=1.0.0-SNAPSHOT, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.file.FileStreamSourceConnector, name='org.apache.kafka.connect.file.FileStreamSourceConnector', version='1.0.0-SNAPSHOT', encodedVersion=1.0.0-SNAPSHOT, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockConnector, name='org.apache.kafka.connect.tools.MockConnector', version='1.0.0-SNAPSHOT', encodedVersion=1.0.0-SNAPSHOT, type=connector, typeName='connector', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSinkConnector, name='org.apache.kafka.connect.tools.MockSinkConnector', version='1.0.0-SNAPSHOT', encodedVersion=1.0.0-SNAPSHOT, type=sink, typeName='sink', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.MockSourceConnector, name='org.apache.kafka.connect.tools.MockSourceConnector', version='1.0.0-SNAPSHOT', encodedVersion=1.0.0-SNAPSHOT, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.SchemaSourceConnector, name='org.apache.kafka.connect.tools.SchemaSourceConnector', version='1.0.0-SNAPSHOT', encodedVersion=1.0.0-SNAPSHOT, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSinkConnector, name='org.apache.kafka.connect.tools.VerifiableSinkConnector', version='1.0.0-SNAPSHOT', encodedVersion=1.0.0-SNAPSHOT, type=source, typeName='source', location='classpath'}, PluginDesc{klass=class org.apache.kafka.connect.tools.VerifiableSourceConnector, name='org.apache.kafka.connect.tools.VerifiableSourceConnector', version='1.0.0-SNAPSHOT', encodedVersion=1.0.0-SNAPSHOT, type=source, typeName='source', location='classpath'}"
}
```

I think this got confusing because I had used `connType` as equivalent to `connClass` because I wasn't using `Type` to differentiate source vs sink, so I had just used `type` and `class` in variable names interchangeably. It might help to convert all the uses of `connType` to `connClass` first, then standardize on class == java class, type == source/sink, name == user-specified name so that even if all you have is a string, you know what it represents.

Reply:
I agree that variable naming is a bit confusing.
Should the renaming be done in another JIRA ?

To get the connector class in this method, we need access to the config Map.
I took a look at ConfigBackingStore but didn't find suitable getter.

Mind giving a bit hint ?

thanks

Reply:
If we can change Herder interface, how about passing the config Map to connectorStatus() ?
```
ConnectorStateInfo connectorStatus(String connName, Map<String, String> config);
```
This way the subclasses pass the Map to AbstractHerder and the problem is solved.

Reply:
ConnectorsResource#getConnectorStatus() would need the config map.

The question becomes whether Herder should expose method for returning the config map.

Reply:
@tedyu Well, both subclasses have `configState` fields, they just didn't make it up into this abstract class. We could either require a protected abstract method to return it and implement it in both, or try to lift the field into this abstract class as a protected field. I think we would just need to be careful re: synchronization w/ using it since the herders need to be thread safe, and this field in particular requires synchronization.

Reply:
By introducing config getter, there is no need to modify the signature of connectorStatus()